routes/shopItem.js

```diff
@@ -1,64 +1,58 @@
 const express = require("express");
-const utility = require("../utility.js");
-const db = require("../server.js");
+const sql = require("../database/database.js");
 
-shopItemRouter = express.Router();
+const shopItemRouter = express.Router();
 
 shopItemRouter.post("/", (req, res, next) => {
```

Comment:
You can use `aysnc` request handlers with express which would make your code shorter and easier to understand:

```js
shopItemRouter.post("/", async (req, res, next) => {
  // some more code...
  const queryGroceries = async  (name, is_completed) => { /* the rest... */ }
  // some more core...
  // instead of queryGroceries(...).then(/* stuff */)
  // you can just
  await queryGroceries(req.body.name, req.body.is_completed);
  // ^- this will just throw on error so you can add back the try - catch from the previous function

  // some more code...
}
```

Same for the rest...

If you need a result you can do:

```js
const data = await myQuery(arg1, arg2);
```

`await` is basically a "syntax sugar":

```js
// this code:
try {
  const someResult = await someAsyncFunc(args);
  doStuff(someResult);
} catch (someError) {
  // some error handling
}

// is pretty much equivalent to:
someAsyncFunc(arg).then(
   (someResult) => {
    doStuff(someResult);
  },
  (someError) => { 
    // some error handling
  });
```